src/Test.UnitTests.Sarif.Multitool/RebaseUriCommandTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Multitool;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Multitool
+{
+    public class RebaseUriCommandTests : FileDiffingUnitTests
+    {
+        private static ResourceExtractor Extractor = new ResourceExtractor(typeof(RebaseUriCommandTests));
+
+        public RebaseUriCommandTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+        [Fact]
+        public void RebaseUriCommand_RebaseRunWithArtifacts()
+        {
+            string testFilePath = "RunWithArtifacts.sarif";
+
+            var options = new RebaseUriOptions
+            {
+                BasePath = @"C:\vs\src\2\s\",
+                BasePathToken = "SRCROOT",
+                Inline = true,
+                SarifOutputVersion = SarifVersion.Current,
+                PrettyPrint = true
+            };
+
+            RunAndValidateRebaseUriCommand(testFilePath, options);
+        }
+
+        private void RunAndValidateRebaseUriCommand(string testFilePath, RebaseUriOptions options)
+        {
+            string inputSarifLog = Extractor.GetResourceText($"RebaseUriCommand.Inputs.{testFilePath}");
+            string expectedOutput = Extractor.GetResourceText($"RebaseUriCommand.ExpectedOutputs.{testFilePath}");
+
+            string logFilePath = @"c:\logs\mylog.sarif";
+            StringBuilder transformedContents = new StringBuilder();
+
+            options.TargetFileSpecifiers = new string[] { logFilePath };
+
+            Mock<IFileSystem> mockFileSystem = ArrangeMockFileSystem(inputSarifLog, logFilePath, transformedContents);
+
+            var rebaseUriCommand = new RebaseUriCommand(mockFileSystem.Object);
+
+            int returnCode = rebaseUriCommand.Run(options);
+            string actualOutput = transformedContents.ToString();
+
+            returnCode.Should().Be(0);
+
+            Assert.True(
+                AreEquivalent<SarifLog>(actualOutput, expectedOutput),
+                GetErrorMessageWithDiffFileDetails(testFilePath, expectedOutput, actualOutput)
```

Comment:
>GetErrorMessageWithDiffFileDetails [](start = 16, length = 34)

This will be called even in the happy path where the test passes, which means we'll write two files to the file system even when the test passes. Is this a consequence of not having the rebaseline functionality? If so, it strengthens my suggestion (below) that you add rebaselining in this PR.

Reply:
oh, thats good feedback, now that i think about it yes - it does perform the unnecessary steps. This statement is gone because we are now using the base method for this goodness, but i will keep this is in mind for future.

---
In reply to: [298673152](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298673152) [](ancestors = 298673152)